static/js/timerender.ts

```diff
@@ -226,8 +308,8 @@ export function render_date(time: Date, time_above: Date | undefined, today: Dat
 
 // Renders the timestamp returned by the <time:> Markdown syntax.
 export function format_markdown_time(time: number | Date): string {
-    const hourformat = user_settings.twenty_four_hour_time ? "HH:mm" : "h:mm a";
-    return format(time, "E, MMM d yyyy, " + hourformat);
+    const format: DateWithTimeFormat = user_settings.twenty_four_hour_time ? "long24" : "long12";
+    return get_localized_date_or_time_for_format(time, format);
```

Comment:
Would it make sense to move the `user_settings.twenty_four_hour_time ? "long24" : "long12"` logic into `get_localized_date_or_time_for_format(time, "long")`?

I'm not aware of a situation where we'd want to format dates not using the correct logic for `user_settings.twenty_four_hour_time`, and it seems like it'd be nice to thus the 12/24 modifiers exist in a single place in that functions, rather than all the callers.

Reply:
If I understand correctly, you are suggesting that we move all the logic related to `user_settings.twenty_four_hour_time` into the `get_localized_date_or_time_for_format` function. If so, we would need to add an argument that configures the time format, i.e. `get_localized_date_or_time_for_format(time, "short")` would return `Jul 27` and `get_localized_date_or_time_for_format(time, "short", {withTime: true})` would return `Jul 27 1:30 AM` or `Jul 27 01:30`, depending on the user's settings. 
In this example I used an object for readability purposes, but it could also be a boolean value if that is more convenient.

Reply:
Actually, I've found a better solution. I think you should take a look at the code. ;)

Reply:
Sounds good!